test/app.e2e-spec.ts

```diff
@@ -1,20 +1,307 @@
 import { Test, TestingModule } from '@nestjs/testing';
-import { INestApplication } from '@nestjs/common';
+import request from 'supertest';
+import { INestApplication, ValidationPipe, HttpStatus } from '@nestjs/common';
 import { AppModule } from './../src/app.module';
+import { DatabaseService } from './../src/database/database.service';
+import { CREATE_USER_DTO_TEST_OBJ, 
+  CREATE_USER_DTO_TEST_OBJ_WITH_UNMATCHING_PASSWORDS, 
+  TEST_USER, CREATE_USER_DTO_TEST_OBJ_WITH_INVALID_EMAIL_FIELD, 
+  CREATE_USER_DTO_TEST_OBJ_WITH_MISSING_PASSWORD_FIELD, 
+  CREATE_USER_DTO_TEST_OBJ_WITH_MISSING_PASSWORD_CONFIRMATION_FIELD, 
+  LOGIN_AUTHENTICATION, UPDATE_USER_DTO_TEST_OBJ_WITH_UPDATED_PASSWORD, 
+  UPDATE_USER_DTO_WITH_MISSING_CURRENT_PASSWORD_FIELD, 
+  UPDATE_USER_DTO_WITH_INVALID_CURRENT_PASSWORD, 
+  DELETE_USER_DTO_TEST_OBJ, 
+  DELETE_FAILURE_USER_DTO_TEST_OBJ, 
+  CREATE_USER_DTO_ADMIN,
+  ADMIN,
+  CREATE_USER_DTO_TEST_OBJ_WITH_MISSING_ROLE,
+  UPDATE_USER_DTO_WITHOUT_PASSWORD_FIELDS
+} from './test.constants';
 
 describe('AppController (e2e)', () => {
   let app: INestApplication;
+  let databaseService: DatabaseService;
 
-  beforeEach(async () => {
+  beforeAll(async () => {
     const moduleFixture: TestingModule = await Test.createTestingModule({
       imports: [AppModule],
+      providers: [DatabaseService]
     }).compile();
 
+    databaseService = moduleFixture.get<DatabaseService>(DatabaseService);
+
     app = moduleFixture.createNestApplication();
+    app.useGlobalPipes(
+      new ValidationPipe({
+        transform: true,
+        whitelist: true
+      })
+    );
     await app.init();
   });
 
-  afterAll(() => {
-    app.close();
+  beforeEach(async () => {
+    await databaseService.cleanAll();
+  });
+
+  describe('/users', () => {
+
+    describe('POST', () => {
+      it('should return 201 status when user is created', async () => {
+        return await request(app.getHttpServer()).post('/users').set('Content-Type', 'application/json').send(CREATE_USER_DTO_TEST_OBJ).expect(HttpStatus.CREATED).then(response => {
```

Comment:
Throughout these tests would it make more sense to have the comparisons be against the DTO that you submitted?